### ISSUE TITLE
Issue while running on next 12

### ISSUE THREAD

**Issue by pkawula** [scottrippey/next-router-mock#47]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `next-router-mock@0.6.9` for the project I'm working on.

The problem is that next does not export `getRouteRegex` and `getRouteMatcher` from theirs `next/dist/shared/lib/router/utils`
I am using the next@canary (18 currently)

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/next-router-mock/dist/dynamic-routes/extensions-11.1.js b/node_modules/next-router-mock/dist/dynamic-routes/extensions-11.1.js
index 71b36de..b05f2b6 100644
--- a/node_modules/next-router-mock/dist/dynamic-routes/extensions-11.1.js
+++ b/node_modules/next-router-mock/dist/dynamic-routes/extensions-11.1.js
@@ -1,6 +1,10 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
-const utils_1 = require("next/dist/shared/lib/router/utils");
+const utils = require("next/dist/shared/lib/router/utils");
+const regex = require("next/dist/shared/lib/router/utils/route-regex");
+const matcher = require("next/dist/shared/lib/router/utils/route-matcher");
+
+const utils_1 = { ...utils, ...regex, ...matcher }
 //
 const normalize_page_path_1 = require("next/dist/server/normalize-page-path");
 const MemoryRouter_registerPaths_1 = require("./MemoryRouter.registerPaths");
diff --git a/node_modules/next-router-mock/dist/dynamic-routes/extensions-12.js b/node_modules/next-router-mock/dist/dynamic-routes/extensions-12.js
index f5f78bd..211337c 100644
--- a/node_modules/next-router-mock/dist/dynamic-routes/extensions-12.js
+++ b/node_modules/next-router-mock/dist/dynamic-routes/extensions-12.js
@@ -1,6 +1,10 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
-const utils_1 = require("next/dist/shared/lib/router/utils");
+const utils = require("next/dist/shared/lib/router/utils");
+const regex = require("next/dist/shared/lib/router/utils/route-regex");
+const matcher = require("next/dist/shared/lib/router/utils/route-matcher");
+
+const utils_1 = { ...utils, ...regex, ...matcher }
 // @ts-ignore
 const normalize_page_path_1 = require("next/dist/shared/lib/page-path/normalize-page-path");
 const MemoryRouter_registerPaths_1 = require("./MemoryRouter.registerPaths");
diff --git a/node_modules/next-router-mock/src/dynamic-routes/extensions-11.1.ts b/node_modules/next-router-mock/src/dynamic-routes/extensions-11.1.ts
index aae89ce..d2e46e0 100644
--- a/node_modules/next-router-mock/src/dynamic-routes/extensions-11.1.ts
+++ b/node_modules/next-router-mock/src/dynamic-routes/extensions-11.1.ts
@@ -1,10 +1,16 @@
 import {
-  getRouteMatcher,
-  getRouteRegex,
   getSortedRoutes,
   isDynamicRoute,
   //
 } from "next/dist/shared/lib/router/utils";
+import {
+  getRouteRegex
+  //
+} from "next/dist/shared/lib/router/utils/route-regex";
+import {
+  getRouteMatcher
+  //
+} from "next/dist/shared/lib/router/utils/route-matcher";
 //
 import { normalizePagePath } from "next/dist/server/normalize-page-path";
 
diff --git a/node_modules/next-router-mock/src/dynamic-routes/extensions-12.ts b/node_modules/next-router-mock/src/dynamic-routes/extensions-12.ts
index 5460ddc..77f0db0 100644
--- a/node_modules/next-router-mock/src/dynamic-routes/extensions-12.ts
+++ b/node_modules/next-router-mock/src/dynamic-routes/extensions-12.ts
@@ -1,10 +1,16 @@
 import {
-  getRouteMatcher,
-  getRouteRegex,
   getSortedRoutes,
   isDynamicRoute,
   //
 } from "next/dist/shared/lib/router/utils";
+import {
+  getRouteRegex
+  //
+} from "next/dist/shared/lib/router/utils/route-regex";
+import {
+  getRouteMatcher
+  //
+} from "next/dist/shared/lib/router/utils/route-matcher";
 // @ts-ignore
 import { normalizePagePath } from "next/dist/shared/lib/page-path/normalize-page-path";
 
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by scottrippey** [comment]
Thanks for reporting this.  I've been trying to figure out how to make this code less brittle, since it's currently dependent on these Next-internal paths.
I'll investigate the canary release, and see what I can do to better support the upcoming changes.

**Comment by scottrippey** [comment]
I was able to add some unit tests to validate the various Next versions, including v10, 11, 12, and canary.   #48 
Also, updated the paths for `getRouteRegex / getRouteMatcher` so it will work with Next Canary #49 

Published as `v0.6.10`.  Please let me know if it works!  Enjoy!
